memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateConversationTransportAction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.memory.action.conversation;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.DocWriteResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.ml.common.conversation.ConversationalIndexConstants;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class UpdateConversationTransportAction extends HandledTransportAction<ActionRequest, UpdateResponse> {
+    Client client;
+
+    @Inject
+    public UpdateConversationTransportAction(TransportService transportService, ActionFilters actionFilters, Client client) {
+        super(UpdateConversationAction.NAME, transportService, actionFilters, UpdateConversationRequest::new);
+        this.client = client;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<UpdateResponse> listener) {
+        UpdateConversationRequest updateConversationRequest = UpdateConversationRequest.fromActionRequest(request);
+        String conversationId = updateConversationRequest.getConversationId();
+        UpdateRequest updateRequest = new UpdateRequest(ConversationalIndexConstants.META_INDEX_NAME, conversationId);
+        updateRequest.doc(updateConversationRequest.getUpdateContent());
+        updateRequest.docAsUpsert(true);
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.update(updateRequest, getUpdateResponseListener(conversationId, listener, context));
+        } catch (Exception e) {
+            log.error("Failed to update Conversation for conversation id" + conversationId, e);
+            listener.onFailure(e);
+        }
+    }
+
+    private ActionListener<UpdateResponse> getUpdateResponseListener(
+        String conversationId,
+        ActionListener<UpdateResponse> actionListener,
+        ThreadContext.StoredContext context
+    ) {
+        return ActionListener.runBefore(ActionListener.wrap(updateResponse -> {
+            if (updateResponse != null && updateResponse.getResult() == DocWriteResponse.Result.UPDATED) {
+                log.info("Successfully updated the Conversation with ID: {}", conversationId);
+                actionListener.onResponse(updateResponse);
+            } else {
+                log.info("Failed to update the Conversation with ID: {}", conversationId);
```

Comment:
In which cases this can happen? During that time don't we expect exception?

Reply:
For example, the client.threadPool().getThreadContext() failed, this catch will catch and log the messages. Anything exception that is not catched in the cmHandler.getTraces() will go here.

Reply:
That part I understand too. I'm more curious to know about `anything exception that` cases. Because any kind of exception we are catching. But in this clause,  exception won't happen but still somehow we won't get the update response. So trying to understand in which cases this can happen?

Reply:
@dhrubo-os  It will be hard to tell and construct edge case for this exception. I think some edge case like out of disk space, then the update will fail, or some circuit breaker like memory triggered may also cause update failure.

Reply:
I see, don't we get exception during those cases too? I'm just wondering, update failed, but we still sending success response with just a log. Is this reasonable or not.

Reply:
This log.error() means the update itself returned error but Ml-Commons finished processing the request successfully because we received the updateResponse. If we get any exceptions during the request, the actionListener will report error back to the customer through actionListener.onFailure(exception).

Reply:
[nit] Let's add `log.error` then instead of `log.info` at least.